### ISSUE TITLE
Automatic Sidecar Injection Not Happening on BookInfo Demo

### ISSUE THREAD

**Issue by davesargrad** [istio/istio#23153]
I've followed the instructions to install bookinfo.

I see my apps but they do not become ready.

Here they are in the coolio namespace.
![image](https://user-images.githubusercontent.com/48793689/80021653-d1f3a800-84a8-11ea-829c-4fe2235b89a9.png)

I've properly labeled that namespace.
![image](https://user-images.githubusercontent.com/48793689/80021712-f059a380-84a8-11ea-921f-865143f7c153.png)

The replicaset description indicates failure to inject sidecar.

![image](https://user-images.githubusercontent.com/48793689/80021900-3d3d7a00-84a9-11ea-9a85-a9ac6435c2b7.png)

Initially I thought my problem was that I was using [flannel](https://github.com/coreos/flannel/issues/1259), as seen in [this ](https://github.com/istio/istio/issues/21058)istio issue. However, I've reset my cluster and am now using calico.
![image](https://user-images.githubusercontent.com/48793689/80021786-0bc4ae80-84a9-11ea-84d8-a5b0ee99d231.png)

**istio-system resources**
```
[dsargrad@ghana istio-1.5.1]$ k get all
NAME                                        READY   STATUS    RESTARTS   AGE
pod/grafana-5cc7f86765-9rs9z                1/1     Running   0          10m
pod/istio-egressgateway-598d7ffc49-prfdh    1/1     Running   0          10m
pod/istio-ingressgateway-7bd5586b79-6szrr   1/1     Running   0          10m
pod/istio-tracing-8584b4d7f9-hdzfl          1/1     Running   0          10m
pod/istiod-646b6fcc6-cs7rx                  1/1     Running   0          10m
pod/kiali-696bb665-ms8vg                    1/1     Running   0          10m
pod/prometheus-6c88c4cb8-5w66q              2/2     Running   0          10m

NAME                                TYPE           CLUSTER-IP       EXTERNAL-IP   PORT(S)                                                                                                                                      AGE
service/grafana                     ClusterIP      10.101.178.57    <none>        3000/TCP                                                                                                                                     10m
service/istio-egressgateway         ClusterIP      10.96.188.133    <none>        80/TCP,443/TCP,15443/TCP                                                                                                                     10m
service/istio-ingressgateway        LoadBalancer   10.105.252.48    <pending>     15020:30454/TCP,80:32350/TCP,443:32695/TCP,15029:32555/TCP,15030:31715/TCP,15031:32420/TCP,15032:30245/TCP,31400:30037/TCP,15443:30217/TCP   10m
service/istio-pilot                 ClusterIP      10.109.165.58    <none>        15010/TCP,15011/TCP,15012/TCP,8080/TCP,15014/TCP,443/TCP                                                                                     10m
service/istiod                      ClusterIP      10.101.240.68    <none>        15012/TCP,443/TCP                                                                                                                            10m
service/jaeger-agent                ClusterIP      None             <none>        5775/UDP,6831/UDP,6832/UDP                                                                                                                   10m
service/jaeger-collector            ClusterIP      10.104.193.120   <none>        14267/TCP,14268/TCP,14250/TCP                                                                                                                10m
service/jaeger-collector-headless   ClusterIP      None             <none>        14250/TCP                                                                                                                                    10m
service/jaeger-query                ClusterIP      10.101.147.121   <none>        16686/TCP                                                                                                                                    10m
service/kiali                       ClusterIP      10.108.178.80    <none>        20001/TCP                                                                                                                                    10m
service/prometheus                  ClusterIP      10.98.118.118    <none>        9090/TCP                                                                                                                                     10m
service/tracing                     ClusterIP      10.109.235.26    <none>        80/TCP                                                                                                                                       10m
service/zipkin                      ClusterIP      10.107.123.27    <none>        9411/TCP                                                                                                                                     10m

NAME                                   READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/grafana                1/1     1            1           10m
deployment.apps/istio-egressgateway    1/1     1            1           10m
deployment.apps/istio-ingressgateway   1/1     1            1           10m
deployment.apps/istio-tracing          1/1     1            1           10m
deployment.apps/istiod                 1/1     1            1           10m
deployment.apps/kiali                  1/1     1            1           10m
deployment.apps/prometheus             1/1     1            1           10m

NAME                                              DESIRED   CURRENT   READY   AGE
replicaset.apps/grafana-5cc7f86765                1         1         1       10m
replicaset.apps/istio-egressgateway-598d7ffc49    1         1         1       10m
replicaset.apps/istio-ingressgateway-7bd5586b79   1         1         1       10m
replicaset.apps/istio-tracing-8584b4d7f9          1         1         1       10m
replicaset.apps/istiod-646b6fcc6                  1         1         1       10m
replicaset.apps/kiali-696bb665                    1         1         1       10m
replicaset.apps/prometheus-6c88c4cb8              1         1         1       10m

```

The istio-pilot ..
![image](https://user-images.githubusercontent.com/48793689/80023040-edf84900-84aa-11ea-93d8-9a3cb605b718.png)

--------------------

I have a fresh k8s cluster. I reset my dev cluster to get calico and istio installed. I did some simple tests to ensure that my cluster is working. It is.

Right now I have a single node cluster, with the master able to run pods. 
![image](https://user-images.githubusercontent.com/48793689/80024385-f6518380-84ac-11ea-8a96-7b34a61e7a65.png)

K8S Version
![image](https://user-images.githubusercontent.com/48793689/80023332-5a734800-84ab-11ea-847b-dc0ce14d4d82.png)

Kernel Version
![image](https://user-images.githubusercontent.com/48793689/80023382-71199f00-84ab-11ea-91a9-abdc74a701ed.png)

OS Version
![image](https://user-images.githubusercontent.com/48793689/80023487-97d7d580-84ab-11ea-9c6a-b4e92cd99054.png)

**Comment by davesargrad** [comment]
Same behaviour if I run bookinfo in default (rather than my own namespace)
![image](https://user-images.githubusercontent.com/48793689/80023884-38c69080-84ac-11ea-91c5-439eb9ea4fc6.png)


![image](https://user-images.githubusercontent.com/48793689/80023936-4b40ca00-84ac-11ea-82de-6a90dd49a091.png)

**Comment by davesargrad** [comment]
I'm hoping someone from the istio team can suggest a couple things for me to look at. I can provide detailed feedback that hopefully will help you to ensure that such things get ironed out. It seems like this kind of failure can lead to slower adoption of istio.

**Comment by davesargrad** [comment]
I have noticed that the istiod pod outputs the following once every second:
![image](https://user-images.githubusercontent.com/48793689/80101038-d5397300-853e-11ea-9c9d-3b5b0aff249c.png)

**Comment by davesargrad** [comment]
I have a simple reader_pod, writer_pod, that use a simple "fortune generator".

This too fails to create, because of the istio sidecar injection failure, as seen in the following.
![image](https://user-images.githubusercontent.com/48793689/80101714-ca331280-853f-11ea-83dc-d1f9d0513c5a.png)

**Comment by esnible** [comment]
I would be curious to know what you see with `istioctl kube-inject -f writer_pod.yaml | kubectl apply -f -`.  Any interesting error messages?

If it fails, for some non-obvious reason, attach the output of `istioctl kube-inject` here and I will investigate.

**Comment by davesargrad** [comment]
Hi Ed @esnible 
Thanks for the response. I have isolated this. I found the solution on this [jetstack issue](https://github.com/jetstack/cert-manager/issues/2640).

It turns out that a corporate proxy creates many challenges for a fully operational k8s cluster. The solution was to add .svc, .svc.cluster.local to my no_proxy environment. I then had to update the kubernetes manifests for kube-apiserver, kube-controllermanager, and kube-scheduler.

I will do further testing to ensure that istio is behaving as one would expect. I'll be glad to share what I learn if I continue to run into such challenges.
